modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/SuggestedFoldersManager.kt

```diff
@@ -0,0 +1,45 @@
+package au.com.shiftyjelly.pocketcasts.repositories.podcast
+
+import au.com.shiftyjelly.pocketcasts.models.db.AppDatabase
+import au.com.shiftyjelly.pocketcasts.models.db.dao.SuggestedFoldersDao
+import au.com.shiftyjelly.pocketcasts.models.entity.SuggestedFolder
+import au.com.shiftyjelly.pocketcasts.servers.podcast.PodcastCacheServiceManager
+import au.com.shiftyjelly.pocketcasts.servers.podcast.SuggestedFoldersRequest
+import jakarta.inject.Inject
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import timber.log.Timber
+
+class SuggestedFoldersManager @Inject constructor(
+    private val podcastCacheService: PodcastCacheServiceManager,
+    appDatabase: AppDatabase,
+) : CoroutineScope {
+
+    private val suggestedFoldersDao: SuggestedFoldersDao = appDatabase.suggestedFoldersDao()
+
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default
+
+    fun getSuggestedFolders(): Flow<List<SuggestedFolder>>? {
+        try {
+            return suggestedFoldersDao.findAll()
+        } catch (e: Exception) {
+            Timber.e(e, "Getting suggested folders failed")
+            return null
+        }
```

Comment:
Can you explain why is there a need for `try/catch` here? Do you expect database to crash while reading from it?

Reply:
I am being conservative here, but I don't expect a crash from the database

Reply:
I'm asking because I don't think we do this in any other places. Maybe there are some single instances. I'm not saying that this is necessarily wrong. I'd even say that it is strictly correct to have `try/catch` when interacting with DB. But if that's something you think is worth doing I think we should discuss it in Slack or during one of our syncs instead of sometimes having it in the code.

Reply:
I don't have strong opinion about this. I will remove it in the next commits